abstractions/java/lib/src/main/java/com/microsoft/kiota/authentication/BaseBearerTokenAuthenticationProvider.java

```diff
@@ -28,10 +28,10 @@ public BaseBearerTokenAuthenticationProvider(@Nonnull final AccessTokenProvider
             }
             return this.accessTokenProvider.getAuthorizationToken(targetUri)
                 .thenApply(token -> {
-                    if(token == null || token.isEmpty()) {
-                        throw new UnsupportedOperationException("Could not get an authorization token", null);
+                    if(!(token == null || token.isEmpty())) { 
```

Comment:
```suggestion
                    if(token != null && !token.isEmpty())) { 
```

Reply:
Maybe it's just me but it took me a while to understand the condition

Reply:
it was a test for your boolean algebra, you passed :P